File: bindings/go/examples/dynamic_fields.go
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package main

import (
	"fmt"
	"log"

	sdk "bindings/iota_sdk_ffi"
)

func isNilError(err error) bool {
	if sdkErr, ok := err.(*sdk.SdkFfiError); ok {
		return sdkErr == nil
	}
	return false
}

func main() {
	client := sdk.GraphQlClientNewDevnet()

	parentObjectId, err := sdk.AddressFromHex("0x07c59b37bd7d036bf78fa30561a2ab9f7a970837487656ec29466e817f879342")
	if err != nil {
		log.Fatalf("Failed to parse address: %v", err)
	}

	pagination := sdk.PaginationFilter{
		Direction: sdk.DirectionForward,
		Cursor:    nil,
		Limit:     nil,
	}

	page, err := client.DynamicFields(parentObjectId, pagination)
	if !isNilError(err) {
		log.Fatalf("Failed to get dynamic fields: %v", err)
	}

	fmt.Printf("Page info: %+v\n", page.PageInfo)
	fmt.Printf("Page size: %d\n", len(page.Data))
	if len(page.Data) > 0 {
		fmt.Printf("First field name: %+v\n", page.Data[0].Name)
		fmt.Printf("First field value: %v\n", *page.Data[0].ValueAsJson)
	}
}

File: bindings/swift/examples/dynamic_fields.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import IotaSDK

@main
struct DynamicFieldsExample {
  static func main() async throws {
    let client = GraphQlClient.newTestnet()
    let parentObjectId = try Address.fromHex(
      hex: "0x7cab491740d51e0d75b26bf9984e49ba2e32a2d0694cabcee605543ed13c7dec")
    let page = try await client.dynamicFields(address: parentObjectId)
    print("Page size:", page.data.count)
    if !page.data.isEmpty {
      print("First field name:\n", page.data[0].name)
      print("First field value:\n", page.data[0].valueAsJson as Any)
    }
  }
}
